### ISSUE TITLE
Add support for Vivaldi Snapshot and Vivaldi Soprano's

### ISSUE THREAD

**Issue by Stephan-P** [bitwarden/mobile#673]
Would it be possible to also add the following supported browsers in https://github.com/bitwarden/mobile/commit/f657edf195d4770c610cf2cd03b88a5f00d5d033
- com.vivaldi.browser.snapshot
- com.vivaldi.browser.sopranos

I know that Vivaldi Team has some stuff to do on their own side, but Boy wouldn't it be a pity if I wouldn't be able to test it because Bitwarden may not be listening?

**Comment by Stephan-P** [comment]
Would wildcards be allowed here? 
- com.vivaldi.browse*
- or an additional com.vivaldi.browser.*

**Comment by contribucious** [comment]
Hello @Stephan-P,

About:

- com.vivaldi.browser.snapshot
  _=> Was added 2 weeks ago in commit https://github.com/bitwarden/mobile/commit/8ad44b405d32a8fcad576084536ff09d7aff56fd from PR https://github.com/bitwarden/mobile/pull/822
  (and in `autofillservice.xml` in commit https://github.com/bitwarden/mobile/commit/41997d5fe0c9556d30cc6e18f1436e226726dbc7 from PR https://github.com/bitwarden/mobile/pull/834)_

- com.vivaldi.browser.sopranos
  _=> See https://github.com/bitwarden/mobile/pull/845 for follow-up (PR pending*, currently)_

**(*)** Note that this issue will be automatically closed once the PR is committed. Comments will still be possible below, however.

___
<details>
  <summary><strong><em>PRESS ME TO KNOW MORE ABOUT VIVALDI SOPRANOS ... :point_left:</em></strong></summary>

### Quotes about it

> Note that there now exist three different versions of Vivaldi Android: **Beta**, **Snapshot** and **Soprano's**. The latter being pre-snapshot build for translators and Vivaldi team members.
> Each of these three versions has its own android app ID.
>
> **Source:** [A Vivaldi translator](https://forum.vivaldi.net/topic/41642/bitwarden-v2-2-6-support-vivaldi-android/13) (in fact ... the author of this issue :relaxed:)

> Our testing process is pretty straightforward. When we finish a new feature, we first test it internally (with the help of a small group of Vivaldi volunteers called the **Sopranos**), then we release the new feature in a **Snapshot**.
>
> **Source:** [Vivaldi](https://vivaldi.com/blog/join-the-ride/)

</details>

**Comment by Stephan-P** [comment]
This sounds wonderful. Is there something that I can do /must do?
I have Vivaldi Sopranos on my Android phone and I'm prepared to switch to another variant for testing purposes.

**Comment by contribucious** [comment]
@Stephan-P Thank you for the proposal. This should however go smoothly¹, but to test in preview if desired, I would say that the easiest way is to join the Beta program (regularly updated²) on the [app page of the Play Store](https://play.google.com/store/apps/details?id=com.x8bit.bitwarden) (which must be open in the Play Store app in this case) and then report any problems encountered here.

___
¹ Using the **accessibility service** at least, because concerning the case of the **_Android Autofill Framework_ in compatibility mode**, there is a [known bug](https://github.com/bitwarden/mobile/issues/489) currently affecting Chromium (it seems) and by that, except maybe a few, the latter's forks. Which will most likely also include Vivaldi Sopranos as a result. But, at least, the part specifically related to this browser is now in place in Bitwarden. :thumbsup:

² However, take into account the slowdown of each validation by Google due to COVID-19 during this period ...

**Comment by Stephan-P** [comment]
I joined the beta programme. Vivaldi sopranos did not connect to Bitwarden beta for autofill, which I believe corresponds with the implementation.
After replacing Vivaldi sopranos with Vivaldi beta, autofill works (through the accessibility service). 
Hip hip hooray!
BTW, Does the Autofill service work better in Android 10. Is there a shed of hope in there?

**Comment by contribucious** [comment]
Note that if Vivaldi Sopranos doesn't work using the "accessibility service", it's **normal** ...
This new modification is most certainly **not yet available** in the Beta program.
_Just wait a few days._

**=>** Each modification made takes a couple of days before being made available via the Play Store (Beta program in this case) unfortunately ... Because to be validated by Google (and during this period, with COVID-19, everything is slowing down ...).

And about improvements for the bug related to Android Autofill Framework, many people are looking forward to better days ... :thumbsup:

**Comment by contribucious** [comment]
@Stephan-P 

I confirm my statements above:

**=>** After checking on my smartphone (participation in the Beta program of Bitwarden), I see that even the addition made [here](https://github.com/bitwarden/mobile/commit/8ad44b405d32a8fcad576084536ff09d7aff56fd) _—the main commit, which includes the support of the accessibility service—_ (and [here](https://github.com/bitwarden/mobile/commit/41997d5fe0c9556d30cc6e18f1436e226726dbc7)) to add support for **Vivaldi Snapshot** 15 days ago has not yet been taken into account there ...

**=>** So the absence there of the _very_ recent support of **Vivaldi Sopranos** makes perfect sense ...

**In summary:** as said, just wait ... :relaxed:

**Comment by Stephan-P** [comment]
Also, Vivaldi Android will be going out of beta today. It's a big day today!
So, I'll probably be changing the sopranos version for the snaphot or even the standard version.

**Comment by Stephan-P** [comment]
Ooh, I'm so happy with the new Vivaldi + Bitwarden beta!
I can't wait to announce this at the Vivaldi community forum after the new release of Bitwarden Mobile is out.

**Comment by contribucious** [comment]
:thumbsup: :relaxed:

**Comment by Stephan-P** [comment]
BTW, isn't it terribly tedious that you have to manually manage a list of supported browser app id's? Is this an Android requirement?

**Comment by contribucious** [comment]
In fact, it's not even enough to maintain a list of these:

regarding the use of the accessibility service for instance, Bitwarden (or any other password manager) also needs to know _how_ to retrieve the URL of the current web page within each browser (to compare with), because the way to do this can vary depending on the browser ... See the [second (and in some cases third) parameter of this list](https://github.com/bitwarden/mobile/blob/master/src/Android/Accessibility/AccessibilityHelpers.cs#L24) in the case of Bitwarden.

**Comment by contribucious** [comment]
@Stephan-P 

By the way, a **new version of Bitwarden beta has been released** recently.
And it includes both **Vivaldi Snapshot** and **Vivaldi Sopranos** support! :relaxed:

Although not having access to the latter to test (being an internal / limited access build), I can suspect it because the commit was made at the same time as the commit adding support for Brave beta (which now works, after checking on my smartphone).

Can you confirm that Vivaldi Sopranos detects websites now instead of "com.vivaldi.browser.sopranos", after updating to the beta "build 2353" (visible in Settings > "About" section)?

**Comment by contribucious** [comment]
@kspearrin In the future, could you consider the possibility of letting us know up to what commit a given "build NNNN" goes? _(related to the post above)_

**Comment by Stephan-P** [comment]
I'd already switched to Vivaldi mainstream, but I've installed Vivaldi Sopranos again to give it a test with Bitwarden beta.

For each login I tried, the Bitwarden overlay popped up, so Bitwarden is in connection with Vivaldi sopranos alright.
However, for each of these logins, Bitwarden beta was not able to find the corresponding item in the database: "no items found for github.org".  I believe this is a new issue of late that also occurs with other browsers.
So, BW took the correct url from the browser, but was not able to match that with database items.

Stephan

**Comment by Stephan-P** [comment]
BW version: 2.3.1 (2353)

**Comment by contribucious** [comment]
OK, perfect, thank you for your feedback. :thumbsup:
And, yes, this is another unrelated problem.

**P.S.** You probably meant **github._com_** I guess. :relaxed:

**Comment by Stephan-P** [comment]
Correction: github.com
